app/models/wallet.rb

```diff
@@ -1,8 +1,22 @@
 class Wallet < ApplicationRecord
+  include WalletFunc
+
+  before_create :set_wallet_number
+
   belongs_to :currency
   belongs_to :user
 
   has_many :transactions, dependent: :nullify
 
   validates :amount, presence: true, numericality: { greater_than_or_equal_to: 0 }
+
+  private
+
+  def set_wallet_number
+    if Wallet.exists?(wallet_number: create_random_wallet_number)
+      set_wallet_number
+    else
+      self.wallet_number = create_random_wallet_number
+    end
+  end
```

Comment:
Here the logic is broken that if a number that already exists comes, it will be generated again and this can happen an infinite number of times. Think about how to rewrite this

Reply:
Imagine that we generate a wallet number = 10
Then there is a check for the existence of such a wallet number, let's say it returns false
Then go to else and again create the wallet number ... 2nd time